modules/nvidia_plugin/src/cuda_operation_base.hpp

```diff
@@ -75,6 +76,8 @@ class OperationBase : public IOperationExec, public IOperationMeta, public std::
                   IndexCollection&& inputIds,
                   IndexCollection&& outputIds);
 
+    bool IsCudaGraphCompatible() const override { return false; }
```

Comment:
You have 6 operations where this value is false and 47 where it is true. Does it really make sense to modify > 100 files to return the same value?

Reply:
We made it this way on purpose, so the default value from the base class is false and all new operations wouldn't support `CudaGraph` by default.
So when a new operation is added, the value would be changed to `true` only when a developer made sure about the `CudaGraph` support of this operation.

Reply:
> We made it this way on purpose, so the default value from the base class is `false` and all new operations wouldn't support `CudaGraph` by default. So when a new operation is added, the value would be changed to `true` only when a developer made sure about the `CudaGraph` support of this operation.

Ok, can we move it into constructor of `OperationBase` with default value `false`?

Reply:
[I.25](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#i25-prefer-empty-abstract-classes-as-interfaces-to-class-hierarchies) rule suggests to prefer empty abstract classes as interfaces to class hierarchies since abstract classes that are empty (have no non-static member data) are more likely to be stable than base classes with state.

Reply:
> [I.25](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#i25-prefer-empty-abstract-classes-as-interfaces-to-class-hierarchies) rule suggests to prefer empty abstract classes as interfaces to class hierarchies since abstract classes that are empty (have no non-static member data) are more likely to be stable than base classes with state.

I didn't suggest modify abstract classes, you can keep inferface empty. I'm talking about `OperationBase` implementation.

Reply:
@nkogteva 
If we add a member value to `OperationBase` and initialize it in the constructors we would need approximately the same amount of changes if the vast majority of operations sets `true` while the default value is `false`.
So the number of changed files would be similar.
But if you think this approach is better, we may do it.

Reply:
> @nkogteva If we add a member value to `OperationBase` and initialize it in the constructors we would need approximately the same amount of changes if the vast majority of operations sets `true` while the default value is `false`. So the number of changed files would be similar. But if you think this approach is better, we may do it.

2x less files will be touched. But it is up yo you, for me it looks like overkill.